lm-agent/tests/parsing/test_flexlm.py

```diff
@@ -3,29 +3,160 @@
 """
 from pytest import mark
 
-from lm_agent.parsing.flexlm import parse
+from lm_agent.parsing.flexlm import parse, parse_feature_line, parse_usage_line
+
+
+def test_parse_feature_line():
+    """
+    Does the regex for the feature line match the lines in the output?
+
+    The line contains:
+    - feature
+    - total
+    - used
+    """
+    assert parse_feature_line(
+        "Users of TESTFEATURE:  (Total of 1000 licenses issued;  Total of 93 licenses in use)"
+    ) == {
+        "feature": "testfeature",
+        "total": 1000,
+        "used": 93,
+    }
+    assert parse_feature_line("not a feature line") is None
+    assert parse_feature_line("") is None
+
+
+def test_parse_usage_line():
+    """
+    Does the regex for the usage line match the line in the output?
+
+    The line contains:
+    - user name
+    - lead host
+    - booked
+    """
+    assert parse_usage_line(
+        "    user1 myserver.example.com /dev/tty (v62.2) (myserver.example.com/24200 12507), start Thu 10/29 8:09, 29 licenses"
+    ) == {
+        "user_name": "user1",
+        "lead_host": "myserver.example.com",
+        "booked": 29,
+    }
+    assert parse_usage_line(
+        "    user2 another.server.com /dev/tty feature=feature (v2023.0) (another.server.com/41020 10223), start Mon 3/11 13:16, 100 licenses"
+    ) == {
+        "user_name": "user2",
+        "lead_host": "another.server.com",
+        "booked": 100,
+    }
+    assert parse_usage_line(
+        "    user3 ER1234 SESOD5045 MSCONE:ADAMS_View (v2023.0331) (alternative.server.com/29065 2639), start Fri 3/8 13:25, 5 licenses"
+    ) == {
+        "user_name": "user3",
+        "lead_host": "ER1234",
+        "booked": 5,
+    }
+    assert parse_usage_line(
+        "    user3 ER1234 SESOD5045 MSCONE:ADAMS_View (v2023.0331) (alternative.server.com/29065 2639), start Fri 3/8 13:25"
+    ) == {
+        "user_name": "user3",
+        "lead_host": "ER1234",
+        "booked": 1,
+    }
+    assert parse_usage_line("aaaaa") is None
```

Comment:
Seems like you could parametrize this test.

Reply:
Done